diff -rupN dosbox-x-dosbox-x-v0.83.12/src/dos/dos_files.cpp dosbox-x-dosbox-x-v0.83.12.patched/src/dos/dos_files.cpp
--- dosbox-x-dosbox-x-v0.83.12/src/dos/dos_files.cpp	2021-04-01 09:54:51.000000000 +0200
+++ dosbox-x-dosbox-x-v0.83.12.patched/src/dos/dos_files.cpp	2021-04-05 09:54:49.221177003 +0200
@@ -1204,7 +1204,7 @@ bool DOS_CreateTempFile(char * const nam
 		for (i=0;i<8;i++) {
 			tempname[i]=(rand()%26)+'A';
 		}
-		tempname[8]='.';tempname[9]='T';tempname[10]='M';tempname[11]='P';tempname[12]=0;
+		tempname[8]=0;
 		//if (DOS_FileExists(name)) {cont=true;continue;} // FIXME: Check name uniqueness
 	} while (cont || (!DOS_CreateFile(name,0,entry) && dos.errorcode==DOSERR_FILE_ALREADY_EXISTS));
 	if (dos.errorcode) return false;
