--- dosbox-x/contrib/linux/com.dosbox_x.DOSBox-X.metainfo.xml.in	2021-05-01 10:34:18.092536288 +0200
+++ dosbox-x-clone/contrib/linux/com.dosbox_x.DOSBox-X.metainfo.xml.in	2021-05-01 10:57:19.161911107 +0200
@@ -76,7 +76,7 @@ DOSBox-X emulates a legacy IBM PC and DO
 		<li>Built-in graphical configuration tool</li>
 		<li>Save and load state with support for save files and up to 100 save slots</li>
 		<li>More DOS commands, and ability to add your own programs to the virtual Z: drive</li>
-		<li>Support for clipboard copy & paste and scalable TrueType fonts (TTF)</li>
+		<li>Support for clipboard copy &amp; paste and scalable TrueType fonts (TTF)</li>
 		<li>Various output options, including output scaling (such as pixel-perfect scaling) and OpenGL shaders</li>
 	</ul>
 	With the help of DOSBox-X, you can run plenty of the old classics that don't run on your new computer!
