diff -rupN dosbox-x-dosbox-x-v0.83.12/contrib/linux/com.dosbox_x.DOSBox-X.metainfo.xml.in dosbox-x-dosbox-x-v0.83.12.patched/contrib/linux/com.dosbox_x.DOSBox-X.metainfo.xml.in
--- dosbox-x-dosbox-x-v0.83.12/contrib/linux/com.dosbox_x.DOSBox-X.metainfo.xml.in	2021-04-01 09:54:51.000000000 +0200
+++ dosbox-x-dosbox-x-v0.83.12.patched/contrib/linux/com.dosbox_x.DOSBox-X.metainfo.xml.in	2021-04-05 10:06:45.358315184 +0200
@@ -57,7 +57,7 @@ Compared to DOSBox, DOSBox-X is much mor
 DOSBox-X emulates a legacy IBM PC and DOS environment, and has many emulation options and features such as:
 </p>
 	<ul>
-		<li>Emulate several PC variants: IBM PC, IBM PCjr, Tandy 1000 and NEC PC-98</li>
+		<li>Emulate several PC variants: IBM PC, IBM PCjr, Tandy 1000, Amstrad and NEC PC-98</li>
 		<li>CPUs: 8086, 286, 386, 486, Pentium, Pentium MMX and Pentium Pro</li>
 		<li>Graphics chipsets: MDA, Hercules, CGA, EGA, MCGA, VGA and SVGA</li>
 		<li>Sound options: Sound Blaster series, Gravis Ultrasound, Disney Sound source and more audio options</li>
@@ -66,7 +66,7 @@ DOSBox-X emulates a legacy IBM PC and DO
 		<li>Printer support and emulation options</li>
 		<li>3dfx Voodoo emulation with Glide wrapper support</li>
 		<li>Network emulation: IPX over UDP, and serial modem</li>
-		<li>Network adapter emulation: NE2000 (not for Flatpak or Snap installs)</li>
+		<li>Network adapter emulation: NE2000 (pcap or slirp backend)</li>
 		<li>Mounting a host directory as a harddisk, floppy or CD-ROM</li>
 		<li>Mounting harddisk images, floppy disk images and CD-ROM images</li>
 		<li>ISO9660, FAT12, FAT16 and FAT32 filesystem support, including long filenames</li>
